### PR TITLE
Fortran2 obsolescence

### PR DESCRIPTION
**Pull request by hammondkd**
**Summary**

Added obsolescence warnings to examples/COUPLE/fortran2 in light of the changes to fortran/lammps.f90.
Also remove the obsolete legacy Fortran 77 wrapper.

**Author(s)**

@hammondkd
Karl Hammond
University of Missouri

**Licensing**

By submitting this pull request, I agree that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

This does not change anything other than comments, but it warns users that future changes may remove this directory.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] A package specific README file has been included or updated
- [x] One or more example input decks are included

### CODE REVIEW

**Comment by akohlmey** [comment]
@hammondkd just added one fix for a bug that I came across earlier today.

There is another issue with the code here: https://github.com/hammondkd/lammps/blob/885108e95ba18415d6ad3939ccead15a596bce69/fortran/lammps.f90#L2813

```
/home/akohlmey/compile/lammps/fortran/lammps.f90:2813:30:

 2813 |         ext_data = [ext_data, fix_external_data()] ! extends ext_data by 1
      |                              1
Error: Fortran 2008: No initializer for allocatable component ‘id’ given in the structure constructor at (1)
make[2]: *** [unittest/fortran/CMakeFiles/flammps.dir/build.make:75: unittest/fortran/CMakeFiles/flammps.dir/home/akohlmey/compile/lammps/fortran/lammps.f90.o] Error 1
```

This is only triggered when compiling testing with full optimization (which is not a good idea most of the time), so it went unnoticed for a while and I am stuck with a bunch of non-Fortran stuff right now...

**Comment by hammondkd** [comment]
> There is another issue with the code here: https://github.com/hammondkd/lammps/blob/885108e95ba18415d6ad3939ccead15a596bce69/fortran/lammps.f90#L2813
> 
> ```
> /home/akohlmey/compile/lammps/fortran/lammps.f90:2813:30:
> 
>  2813 |         ext_data = [ext_data, fix_external_data()] ! extends ext_data by 1
>       |                              1
> Error: Fortran 2008: No initializer for allocatable component ‘id’ given in the structure constructor at (1)
> make[2]: *** [unittest/fortran/CMakeFiles/flammps.dir/build.make:75: unittest/fortran/CMakeFiles/flammps.dir/home/akohlmey/compile/lammps/fortran/lammps.f90.o] Error 1
> ```
> 
> This is only triggered when compiling testing with full optimization (which is not a good idea most of the time), so it went unnoticed for a while and I am stuck with a bunch of non-Fortran stuff right now...

I think I fixed this by creating a constructor that explicitly assigns the allocatable string. The issue was with the Fortran 2003 standard; using -std=f2008 eliminated the warning, which is probably why we didn't see it before.

**Comment by akohlmey** [comment]
> I think I fixed this by creating a constructor that explicitly assigns the allocatable string. The issue was with the Fortran 2003 standard; using -std=f2008 eliminated the warning, which is probably why we didn't see it before.

Thanks. I hope to find some time in the next couple of weeks to revisit the library interface and write more examples and tests and check out the details.

**Comment by akohlmey** [comment]
While looking through the COUPLE examples folder I noticed that it is time to remove the legacy Fortran 77 wrapper and make a few more updates to README and manual files.

File: doc/src/Fortran.rst [hunk 1]
```diff
@@ -5,46 +5,52 @@ The :f:mod:`LIBLAMMPS` module provides an interface to call LAMMPS from
...
 If the LAMMPS library itself has been compiled with MPI support, the
 resulting executable will still be able to run LAMMPS in parallel with
 ``mpirun``, ``mpiexec``, or equivalent.  Please also note that the order
 of the source files matters: the ``lammps.f90`` file needs to be
-compiled first, since it provides the :f:mod:`LIBLAMMPS` module that is
-imported by the Fortran code that uses the interface.  A working example
-can be found together with equivalent examples in C and C++ in the
-``examples/COUPLE/simple`` folder of the LAMMPS distribution.
+compiled first, since it provides the :f:mod:`LIBLAMMPS` module that
+would need to b imported by the Fortran code that uses the interface.  A
```
Comment (stanmoore1) [inline doc/src/Fortran.rst]: typo: b --> be

**Review by stanmoore1 — CHANGES_REQUESTED** [review]

**Review by akohlmey — APPROVED** [review]

**Review by stanmoore1 — APPROVED** [review]

**Pull request merged**